feature/onboarding/permissions/src/main/kotlin/app/k9mail/feature/onboarding/permissions/domain/usecase/CheckPermission.kt

```diff
@@ -0,0 +1,41 @@
+package app.k9mail.feature.onboarding.permissions.domain.usecase
+
+import android.Manifest
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import androidx.core.content.ContextCompat
+import app.k9mail.feature.onboarding.permissions.domain.PermissionsDomainContract.UseCase
+import app.k9mail.feature.onboarding.permissions.domain.entity.Permission
+import app.k9mail.feature.onboarding.permissions.domain.entity.PermissionState
+
+/**
+ * Checks if a [Permission] has been granted to the app.
+ */
+class CheckPermission(
+    private val context: Context,
+) : UseCase.CheckPermission {
```

Comment:
Following a strict definition of use cases, no Android specific code is allowed here and should be delegated to a repository or data source.

Reply:
OBE due to d20206a5b53a01ba18c6710e46fefba48f0b2f94?

Reply:
That's even better